submit final de la encuesta, el alumno completa la encuesta tradicion…

patriciotella opened the pull request:
…al solo con las materias que no aprobo y selecciona las opciones disponibles.

patriciotella committed:
submit final de la encuesta, el alumno completa la encuesta tradicional solo con las materias que no aprobo y selecciona las opciones disponibles.

patriciotella committed:
submit final de la encuesta, el alumno completa la encuesta tradicional solo con las materias que no aprobo y selecciona las opciones disponibles.

romanrizzi commented on src/actions.js:
```diff
@@ -7,6 +7,6 @@ const BACKEND_URL = process.env.BACKEND_URL || 'http://localhost:3000'
   return dispatch => {
-    post(`${BACKEND_URL}/surveys`, survey).then((data) => { console.log(data) })
+    post(`/surveys`, survey).then((data) => { console.log(data) })
   }
```
No es necesario hacer un dispatch con redux

patriciotella replied:
no elimine nada del codigo que estaba, solamente no lo use. Creo que @jcalvento estaba usandolo tambien

jcalvento replied:
Yo prefiero encapsular las cosas así que empezar a mezclar las cosas que irían en los reducers en los componentes. Igual me parece que cualquiera de las dos formas no es bloqueante

romanrizzi replied:
Preferible extraerlo fuera del componente como una función común en mi opinion.

romanrizzi replied:
No se si quedo colgado esto

patriciotella replied:
jajaja, yo dije que no lo estaba usando, pero pense que juli si, por eso no elimine la clase

romanrizzi commented on src/components/SubjectsList.jsx:
```diff
@@ -1,51 +1,62 @@
+    continuePoll = (event) => {
+        this.setState({readyToFinalPool: true})
+        event.preventDefault();
+    };
```
Esto no va aca, lo tenemos que hacer mediante el router a esto. Fijate que yo habia creado una URL para la encuesta

patriciotella committed:
Fix del pr:

Se utiliza el router para llegar a la encuesta final
Se unifican los post, eliminando redux cuando no es necesario

patriciotella committed:
Fix del pr:

Se utiliza el router para llegar a la encuesta final
Se unifican los post, eliminando redux cuando no es necesario

jcalvento commented on src/components/SubjectsList.jsx:
```diff
@@ -2,50 +2,51 @@ import React, { Component } from 'react';
 const swapClasses = (element, classRemove, classAdd) => {
-  element.classList.add(classAdd);
-  element.classList.remove(classRemove);
+    element.classList.add(classAdd);
```
2 espacios de identación ftw

patriciotella replied:
autoidentacion

jcalvento commented on src/components/SubjectsList.jsx:
```diff
@@ -2,50 +2,51 @@ import React, { Component } from 'react';
+                        Haz click en las materias que ya aprobaste, esto evitara que aparezcan en encuestas posteriores.
+                    </div>
+                    <div className="row">
+                        { this.state.subjects.map((name, id) => {
```
Puede ser una función

patriciotella replied:
ya esta asi en master pero lo extraigo

jcalvento commented on src/components/SubjectsPoll.jsx:
```diff
@@ -0,0 +1,86 @@
+                        Ahora decinos que materias tenes pensado cursar!
+                    </div>
+                    <div className="row">
+                        { this.state.subjects.map(subject => {
```
Puede ser una función también

patriciotella replied:
lo extraigo x2

jcalvento commented on src/components/SubjectsPoll.jsx:
```diff
@@ -0,0 +1,86 @@
+    castToSubjectPoll = (data) => {
+        const subjectsPoll = this.state.subjects;
+        Object.values(data).map(subject => {
+            var subjectCast = { id: subject[1], name: subject[0], condition: ''}
```
Estos son los subjects que vienen del back? Para qué es necesario castearlos así? No se puede usar el JSON como viene con las propiedades?

patriciotella replied:
estas eran las materias aprobadas no estaba el endpoint para las encuesta, ahora ya se pueden obtener con el commit de @romanrizzi

patriciotella committed:
Fix del pr:

Se extrae comportamiento a funciones
Se integra el endpoint para obtener las materias de la encuesta (asociado con el id del usuario)

patriciotella committed:
Fix del pr:

Se extrae comportamiento a funciones
Se integra el endpoint para obtener las materias de la encuesta (asociado con el id del usuario)

romanrizzi commented on src/actions/auth.js:
```diff
@@ -8,4 +8,8 @@ const enrollStudent = (studentData, success, error) => {
\ No newline at end of file
+const submitSurvey = (survey, success, error) => {
+  post('/surveys', survey).then(success).catch(error)
+};
```
Esto no va en este archivo, no esta relacionado con auth

romanrizzi commented on src/actions/subjects.js:
```diff
@@ -2,4 +2,8 @@ import { get } from '../httpMethods.js';
 
-export { allSubjects };
\ No newline at end of file
+const nonApproveSubjects = (userId, success, error) => {
```
No seria mejor llamarlo `surveyOptions`?

romanrizzi commented on src/components/SubjectsPoll.jsx:
```diff
@@ -0,0 +1,91 @@
+export default class  SubjectsPoll extends Component {
+    state = { subjects: [] }
+    componentDidMount() {
+        const userId = this.props.location.pathname.split('/');
```
Fijate esto: https://stackoverflow.com/a/45599016

patriciotella replied:
evalue lo que me venia en las props, habia hecho algo como eso al principio, pero no lo tenia :(

romanrizzi commented on src/components/SubjectsPoll.jsx:
```diff
@@ -0,0 +1,91 @@
+        nonApproveSubjects(
+            userId[userId.length-1],
+            data => this.setState({ userId: userId[userId.length-1], subjects: data }))
+    }
```
La forma de accerde al userId es mas facíl usando la api del router (fijate lo que puse arriba)

romanrizzi commented on src/components/SubjectsPoll.jsx:
```diff
@@ -0,0 +1,91 @@
+    constructor(props) {
+        super(props)
+        this.handleSubmit = this.handleSubmit.bind(this)
+    }
```
Esto no hace falta ya que las fat arrow functions (=>) ya bindean el contexto. No hace falta escribir el constructor

patriciotella committed:
Merge branch 'submit_final_encuesta_con_opcion_seleccionada' of https://github.com/arq1-rizzi-tella-calvento/react-frontend into submit_final_encuesta_con_opcion_seleccionada
